### PR TITLE
Add notice for required plugins

### PR DESCRIPTION
**Pull request by NaveenKharwar**
fixes #4

### CODE REVIEW

**Review by ajitbohra — CHANGES_REQUESTED** [review]
seems like accidental PR we are deleting the plugin file!

**Comment by NaveenKharwar** [comment]
> seems like accidental PR we are deleting the plugin file!

Yes, I fixed it now you can review.

File: woocommerce-stats-by-currency.php [hunk 1]
```diff
@@ -124,6 +124,26 @@ function wsbc_add_stats() {
...
+
 }
 add_action( 'woocommerce_after_dashboard_status_widget', 'wsbc_add_stats' );
+
+
+// Admin Notice
+add_action( 'admin_init', 'wsbc_required_plugins' );
+
+function wsbc_required_plugins() {
+    if ( is_admin() && current_user_can( 'activate_plugins' ) &&  !is_plugin_active( 'woocommerce/woocommerce.php' )  ) {
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]: - `admin_init` hooks will only run on admin so `is_admin()` check not required
- only users with `activate_plugins` can activate plugins `current_user_can( 'activate_plugins' )` check here redundant
- We are only checking for `WooCommerce` other dependent plugin `WooCommerce Price By Country` is missing

File: woocommerce-stats-by-currency.php [hunk 2]
```diff
@@ -124,6 +124,26 @@ function wsbc_add_stats() {
...
+        add_action( 'admin_notices', 'wsbc_notice' );
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]: We can use `wp_die( 'Plugins required msg' )` to simplify this

File: woocommerce-stats-by-currency.php [hunk 3]
```diff
@@ -124,6 +124,26 @@ function wsbc_add_stats() {
 			<?php
 		}
 	}
-	
+
 }
 add_action( 'woocommerce_after_dashboard_status_widget', 'wsbc_add_stats' );
+
+
+// Admin Notice
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]: Good if we can follow comment style 

```
/**
* Function description
*/
```

**Review by ajitbohra — CHANGES_REQUESTED** [review]
We can simplify this

``` php
/**
 * Check required plugins
 *
 * @return void
 */
function wsbc_required_plugins() {
	$activated = (array) get_option( 'active_plugins', array() );
	$required  = array(
		'woocommerce/woocommerce.php',
		'woocommerce-product-price-based-on-countries/woocommerce-product-price-based-on-countries.php'
	);

	$required_plugins_activated = count( array_intersect($activated, $required) ) == count( $required );

    if ( !$required_plugins_activated  ) {
		$msg_text    = 'Sorry, This plugin requires <strong>WooCommerce</strong> and <strong>WooCommerce Price Based on Country</strong> plugins to work.';
		$plugins_link   = sprintf( '<a href="%s">Install Plugins</a>', admin_url('plugins.php') );
		$dashboard_link = sprintf( '<a href="%s">Dashboard</a>', admin_url() );
		$msg            = sprintf( '%s <br/><br/> %s | %s', $msg_text, $plugins_link, $dashboard_link );

		wp_die( $msg );
    }
}
register_activation_hook( __FILE__, 'wsbc_required_plugins' );

```

File: woocommerce-stats-by-currency.php [hunk 4]
```diff
@@ -124,6 +124,70 @@ function wsbc_add_stats() {
...
+
+    if ( !$required_plugins_activated  ) { ?>
+   
+        <?php
+        $style          = '<style type="text/css">
+                                .wp-die-message {
+                                    border: 1px solid #ccd0d4;
+                                    border-left: solid 5px #dc3232;
+                                    padding: 12px 0 12px 25px;
+                                }
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]: let's simplify this for better design & style. moving border to body/ error wrapper

```suggestion
                                #error-page {
                                    border-left: solid 5px #dc3232;
                                }
```

Reference Design

die_msg

File: woocommerce-stats-by-currency.php [hunk 5]
```diff
@@ -124,6 +124,70 @@ function wsbc_add_stats() {
...
+        $msg_text       = '<p>Sorry, This plugin requires <strong>WooCommerce</strong> and <strong>WooCommerce Price Based on Country</strong> plugins to work. Learn how to install <strong>WooCommerce Stats By Currency</strong> in the <a href="https://lubus.in/blog/woocommerce-stats-by-currency#guide" target="_blank">guide</a>.</p>';
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]: Currently, the plugin doesn't require setup guide and we don't have one. so let's remove `Learn how to install WooCommerce Stats By Currency in the guide.`

**Review by ajitbohra — CHANGES_REQUESTED** [review]
Also, code feels a bit uncomfortable we can extract CSS and HTML into a separate `HTML view` to simplifying this.

File: woocommerce-stats-by-currency.php [hunk 6]
```diff
@@ -133,3 +133,67 @@ function wsbc_add_stats() {
...
 add_action( 'woocommerce_after_dashboard_status_widget', 'wsbc_add_stats' );
+
+/**
+ * Check required plugins
+ *
+ * @return void
+ */
+
+function wsbc_required_plugins() {
+	$activated = (array) get_option( 'active_plugins', array() );
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]:
```suggestion
	$activated = apply_filters( 'active_plugins', (array) get_option( 'active_plugins', array() ) );
```

Let's respect the data filtering.

File: includes/view/notice.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
+<?php
+
```
Comment (ajitbohra) [inline includes/view/notice.php]: Additional space

File: woocommerce-stats-by-currency.php [hunk 8]
```diff
@@ -133,3 +133,30 @@ function wsbc_add_stats() {
...
+/**
+ * Check required plugins
+ *
+ * @return void
+ */
+
+function wsbc_required_plugins() {
+	$activated = apply_filters( 'active_plugins', (array) get_option( 'active_plugins', array() ) );
+	$required  = array(
+		'woocommerce/woocommerce.php',
```
Comment (ajitbohra) [inline woocommerce-stats-by-currency.php]: No need to check for `woocommerce` as `woocommerce-product-price-based-on-countries` will have it as a dependency.

File: includes/view/notice.php [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    .primary:hover {
+        background: #0071a1;
+        border-color: #0071a1;
+        color: #fff;
+    }
+</style>
+<div class="wsbc-notice">
+    <p>Sorry, This plugin requires <strong>WooCommerce</strong> and <strong>WooCommerce Price Based on Country</strong> plugins installed and activated to work.</p>
```
Comment (ajitbohra) [inline includes/view/notice.php]: No need to check for woocommerce as woocommerce-product-price-based-on-countries will have it as a dependency.

**Review by ajitbohra — CHANGES_REQUESTED** [review]

**Comment by ajitbohra** [comment]
@NaveenKharwar last few changes and we are good to merge this :)

File: includes/view/notice.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    .primary:hover {
+        background: #0071a1;
+        border-color: #0071a1;
+        color: #fff;
+    }
+</style>
+<div class="wsbc-notice">
+    <p>Sorry, This plugin requires <strong>WooCommerce Price Based on Country</strong> plugins installed and activated to work.</p>
```
Comment (ajitbohra) [inline includes/view/notice.php]: Sting here is not translatable!

File: includes/view/notice.php [hunk 11]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Notice output 
+ * 
+ * @package WSBC
+ */
+
+$adminurl = admin_url();
+$pluginpage = admin_url('plugins.php');
```
Comment (ajitbohra) [inline includes/view/notice.php]: use snake case for the name

```suggestion
$dashboard_url = admin_url();
$plugins_url   = admin_url('plugins.php');
```

**Review by ajitbohra — CHANGES_REQUESTED** [review]

**Pull request merged**